src/vs/workbench/contrib/terminal/browser/media/shellIntegration-fish.fish

```diff
@@ -0,0 +1,91 @@
+# ----------------------------------------------------------------------------
+#   Visual Studio Code terminal integration for fish
+#   Copyright (c) Microsoft Corporation. All rights reserved.
+#   Licensed under the MIT License. See License.txt in the project root for license information.
+# ----------------------------------------------------------------------------
+# Manual installation:
+#
+#   (1) Add the following to the end of `$__fish_config_dir/config.fish`:
+#
+#         string match -q "$TERM_PROGRAM" "vscode"
+#         and . (code --locate-shell-integration-path fish)
+#
+#   (2) Restart fish.
+# ----------------------------------------------------------------------------
+# TODO: Confirm all escape sequences once they are finalized.
+# See microsoft/vscode#155639 and microsoft/vscode#139400 for discussion.
+# ----------------------------------------------------------------------------
+
+# Don't run in scripts, other terminals, or more than once per session.
+status is-interactive
+and string match --quiet "$TERM_PROGRAM" "vscode"
+and ! set --query VSCODE_SHELL_INTEGRATION
+or return
+
+set --global VSCODE_SHELL_INTEGRATION 1
+
+# Helper function
+function __vsc_esc -d "Emit escape sequences for VS Code shell integration"
+	builtin printf "\e]633;%s\007" (string join ";" $argv)
+end
+
+# Sent right before executing an interactive command.
+# Marks the beginning of command output.
+function __vsc_cmd_executed --on-event fish_preexec
+	__vsc_esc C
+	__vsc_esc E "$argv"
```

Comment:
Right now multi-line commands won't come through properly:

<img width="441" alt="Screen Shot 2022-08-08 at 3 32 43 pm" src="https://user-images.githubusercontent.com/2193314/183526550-bd2150b0-407a-467a-9f21-c5eb23bb487c.png">


We need to serialize them similar to pwsh to get multi-line commands:

```pwsh
$CommandLine.Replace("\", "\\").Replace("`n", "\x0a").Replace(";", "\x3b")
```

<img width="277" alt="Screen Shot 2022-08-08 at 3 39 48 pm" src="https://user-images.githubusercontent.com/2193314/183526582-36e851b6-5be1-4a12-8697-0dcfc63de05f.png">

We can defer this too as we don't even do it on bash or zsh yet (https://github.com/microsoft/vscode/issues/157546).

Reply:
I think I got this working! (Inasmuch as I can replicate your second screenshot, at least.)